fastscanner/benchmarks/benchmark_redis_read.py

```diff
@@ -66,24 +66,51 @@ async def monitor_batch_timeout():
         await asyncio.sleep(1)
         now = time.time()
 
-        if (
-            batch_start_time
-            and last_received_time
-            and (now - last_received_time) > NO_DATA_TIMEOUT
-        ):
+        if last_received_time and (now - last_received_time) > NO_DATA_TIMEOUT:
             logger.info(
-                f"\nBatch started at: {datetime.fromtimestamp(batch_start_time).strftime('%M:%S.%f')}"
+                f"No messages received for {NO_DATA_TIMEOUT} seconds at: {datetime.fromtimestamp(now).strftime('%M:%S.%f')[:-3]}"
             )
-            logger.info(
-                f"Batch ended at:   {datetime.fromtimestamp(now).strftime('%M:%S.%f')}"
-            )
-            logger.info(f"Total messages read: {total_messages}")
-            logger.info(f"Batch duration: {now - batch_start_time:.6f} seconds\n")
+            if batch_start_time:
+                batch_duration = last_received_time - batch_start_time
+                logger.info(f"\nBatch Summary:")
+                logger.info(
+                    f"First message timestamp: {datetime.fromtimestamp(batch_start_time).strftime('%H:%M:%S.%f')[:-3]}"
+                )
+                logger.info(
+                    f"Last message timestamp:  {datetime.fromtimestamp(last_received_time).strftime('%H:%M:%S.%f')[:-3]}"
+                )
+                logger.info(f"Total messages read: {total_messages}")
+                logger.info(f"Batch duration: {batch_duration:.6f} seconds\n")
             batch_start_time = None
             last_received_time = None
             total_messages = 0
 
 
+async def report_stats():
```

Comment:
What is the goal of this second report_stats? Isn't it enough with the monitor_batch_timeout?